Dockerfile

```diff
@@ -38,16 +42,29 @@ ENV PATH=$VERDACCIO_APPDIR/docker-bin:$PATH \
 
 WORKDIR $VERDACCIO_APPDIR
 
+# https://github.com/Yelp/dumb-init
 RUN apk --no-cache add openssl dumb-init
 
 RUN mkdir -p /verdaccio/storage /verdaccio/plugins /verdaccio/conf
 
-COPY --from=builder /opt/verdaccio-build .
+COPY --from=builder /opt/tarball .
+
+USER root
+# install verdaccio as a global package so is fully handled by npm
+# ensure none dependency is being missing and is prod by default
+RUN npm install -g $VERDACCIO_APPDIR/verdaccio.tgz \ 
+    ## clean up cache
+    && npm cache clean --force \
+    && rm -Rf .npm/ \
+    && rm $VERDACCIO_APPDIR/verdaccio.tgz \
+    # yarn is not need it after this step
+    && rm -Rf /opt/yarn-v1.22.19/
 
 ADD conf/docker.yaml /verdaccio/conf/config.yaml
+ADD docker-bin $VERDACCIO_APPDIR/docker-bin
 
 RUN adduser -u $VERDACCIO_USER_UID -S -D -h $VERDACCIO_APPDIR -g "$VERDACCIO_USER_NAME user" -s /sbin/nologin $VERDACCIO_USER_NAME && \
-    chmod -R +x $VERDACCIO_APPDIR/bin $VERDACCIO_APPDIR/docker-bin && \
+    chmod -R +x /usr/local/lib/node_modules/verdaccio/bin/verdaccio $VERDACCIO_APPDIR/docker-bin && \
```

Comment:
This is actually not having any effect `chmod -R +x /usr/local/lib/node_modules/verdaccio/bin/verdaccio` but doe sit on `$VERDACCIO_APPDIR/docker-bin`